lib/tasks/update_supplier_for_moves.rake

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+namespace :moves do
+  desc 'Update the suppliers for all the moves based on the effective dates of the Suppliers'
+  task update_suppliers: :environment do
+    total = Move.count
+
+    i = 0
+
+    puts 'Updating suppliers for moves...'
+    Move.find_each(batch_size: 200) do |move|
+      move.determine_supplier # TODO: this might need to change
+
+      if ENV['confirm_save'] == 'true'
+        move.save!
+      else # Simulation mode
+        puts("Move: #{move.id} is not valid!") unless move.valid?
+      end
+
+      if (total % 200).zero?
```

Comment:
I don't follow the logic of this - `total` value doesn't change, so why evaluate this expression for every move?

Reply:
good catch,
that's should be  `if (i % 200).zero?`
thanks

Reply:
I don't think that quite works either, as `i` is only incremented inside this check

Reply:
Try something like this

```ruby
i += 1
if (i % 200).zero?
  puts "Updated #{i} out of #{total} moves..."
end
```